reconcile/utils/rest_api_base.py

```diff
@@ -0,0 +1,92 @@
+import threading
+from typing import Any
+from urllib.parse import urljoin
+
+import requests
+
+
+def get_next_url(links: dict[str, dict[str, str]]) -> str | None:
+    """Parse response header 'Link' attribute and return the next page url if exists.
+
+    See
+    * https://gitlab.com/glitchtip/glitchtip-backend/-/blob/master/glitchtip/pagination.py#L34
+    * https://requests.readthedocs.io/en/latest/api/?highlight=links#requests.Response.links
+    """
+    if links.get("next", {}).get("results", "false") == "true":
+        return links["next"]["url"]
+    return None
+
+
+class ApiBase:
+    """This class provides a common standard for REST API clients."""
```

Comment:
Do you want to make this as the base for all api clients we use, I have a similar one can benefit from the base, https://github.com/app-sre/qontract-reconcile/blob/master/tools/cli_commands/cost_report/cost_management_api.py, but it requires oauth token handling and customized retry (backoff and status)

Reply:
I've added `session` and `auth` attributes to `__init__` to customize those things easily.

Reply:
Thank you! I'll try to use the new base for cost  management api!